include/multigrid_pclomp/multi_voxel_grid_covariance_omp_impl.hpp

```diff
@@ -56,6 +56,77 @@
 #include <pcl/filters/boost.h>
 #include "multigrid_pclomp/multi_voxel_grid_covariance_omp.h"
 
+template <typename PointT>
+pclomp::MultiVoxelGridCovariance<PointT>::MultiVoxelGridCovariance(const MultiVoxelGridCovariance& other)
+: pcl::VoxelGrid<PointT>(other),
+  leaves_(other.leaves_),
+  grid_leaves_(other.grid_leaves_),
+  leaf_indices_(other.leaf_indices_),
+  kdtree_(other.kdtree_)
```

Comment:
One my concern is that the `kdtree_` has some `shared_ptr`s. Is this copy shallow copy?
https://github.com/PointCloudLibrary/pcl/blob/1033d24b750eed4f22c9c1c7711398aba7877055/kdtree/include/pcl/kdtree/kdtree_flann.h#L288-L294

Reply:
I will make a note of what anhnv3991 told me.

(1) `flann_index_` is reset each time setInputCloud is used, so the target will change at that time.
https://github.com/PointCloudLibrary/pcl/blob/1033d24b750eed4f22c9c1c7711398aba7877055/kdtree/include/pcl/kdtree/impl/kdtree_flann.hpp#L131-L135

(2) `cloud_` is reset with convertCloudToArray. convertCloudToArray is also called in setInputCloud.
https://github.com/PointCloudLibrary/pcl/blob/1033d24b750eed4f22c9c1c7711398aba7877055/kdtree/include/pcl/kdtree/impl/kdtree_flann.hpp#L428-L498

Therefore, it is considered that there is no practical problem with shallow copying.